### ISSUE TITLE
Dropdowns do not appear intermittently on iOS

### ISSUE THREAD

**Issue by ocdtrekkie** [babybuddy/babybuddy-sandstorm#3]
Alright, so I'm still trying to nail down what could cause this. We now have a kid and can now test this app thoroughly! We have Sandstorm added to homescreen, so this would be in iOS Safari, with BabyBuddy in a Sandstorm iframe.

My wife reported she could not add a feeding or diaper change because the dropdowns would not populate and could not be changed from default/null selections. Thought it was weird and saw some artifacts like the dropdowns persisting after leaving the page. I closed the app and reopened it and things worked fine.

I then returned to my own phone, which had worked fine, doing the same thing for a bit, until I once again closed the app.

My best guess so far is that it happens if you had the app open already on Sandstorm and then return to it without closing it in between. It's very possible this could be caused by something like the Sandstorm grain being closed but the app front end not noticing.

**Comment by cdubz** [comment]
I haven’t taken the time to do any digging on this yet but I don’t think it’s specific to Sandstorm. We haven’t used Baby Buddy for a couple of years now but we do have another custom app I made for calorie tracking that we use daily and it has also had this issue _since the iOS 15 upgrade_. I’ve been meaning to look in to it for along time but keep forgetting. Searching for reference to this issue and iOS 15 may yield something useful.

**Comment by ocdtrekkie** [comment]
Good to know. We definitely both are on iOS 15.

**Comment by cdubz** [comment]
Did a little searching just now but haven’t turned anything particularly relevant —

- Someone reporting a similar issue in an iOS 15 beta (with an app-based web view): https://developer.apple.com/forums/thread/689901
- An interesting note (apparently from Apple directly) about the `select` element UI being changed in iOS 15 (not super helpful here but adds to the iOS connection): https://stackoverflow.com/a/68041387

**Comment by ocdtrekkie** [comment]
Intriguing. I wonder if it will "fix itself" in a future iOS update. If you do end up discovering a code fix for your other app, this is I guess an FYI that it'd be appreciated here as well, though we are working around the issue just fine. App works great for tracking what we're tracking so far, and it's wife-approved, which is rarely the case for the self-hosted apps I propose using!

**Comment by cdubz** [comment]
Yeah, of course. I’ve been testing my other app lately in Safari directly (i.e. not a webapp view) and can’t reproduce the issue so far. At some point I’ll try to get a simple POC that I can use to report this… somewhere at Apple. Seems like it’ll be hard to get a truly reproducible example of the big though.

**Comment by cdubz** [comment]
FWIW here is another reference for the exact issue --

- https://stackoverflow.com/questions/70292301/ios-native-controls-invisible-hidden-in-pwa
- https://developer.apple.com/forums/thread/696346

I don't think there are any feasible workaround options here so we just need to hope that Apple fixes this 🤷 

@ocdtrekkie can you confirm this is the issue you are experiencing? If so I'm going to move this issue to the primary babybuddy repo.

**Comment by ocdtrekkie** [comment]
This sounds accurate. I have managed to accidentally change a selection when the select field isn't showing... the claim that it's there but invisible makes sense to me, and explains how that happened.

If it's just a general PWA-on-iOS issue, there's probably nothing to fix on the app side, I'd agree.
